src/broadcast/metadatabroadcast.cpp

```diff
@@ -0,0 +1,88 @@
+
+#include <QLinkedListIterator>
+
+#include "broadcast/metadatabroadcast.h"
+#include "mixer/playerinfo.h"
+
+MetadataBroadcaster::MetadataBroadcaster()
+        :  m_gracePeriodSeconds(5*60) {
+
+}
+
+void MetadataBroadcaster::slotAttemptScrobble(TrackPointer pTrack) {
+    for (auto it = m_trackedTracks.begin();
+         it != m_trackedTracks.end();
+         ++it) {
+        if (*it == GracePeriod(0, pTrack)) {
+            GracePeriod &trackPeriod = *it;
+            if (trackPeriod.hasBeenEjected &&
+                trackPeriod.m_msElapsed > 
+                m_gracePeriodSeconds*1000.0 ||
+                trackPeriod.firstTimeLoaded) {
+                for (auto &service : m_scrobblingServices) {
```

Comment:
Use universal reference, for non const for loops: auto&& and const auto& for const loops. If the container is an implicit shared qt container you need also to ad as_const cast to avoid a deep copy of the container.

We allign & && and * with the type (no space) please check the whole PR for this.